composer.json

```diff
@@ -9,17 +9,17 @@
         }
     ],
     "require": {
-        "php": "^7.2 || ^8.0",
+        "php": "^8.1",
         "nikic/php-parser": "^3.0 || ^4.0",
-        "symfony/finder": "^3.4 || ^4.4 || ^5.0 || ^6.0",
+        "symfony/finder": "^3.4 || ^4.4 || ^5.0 || ^6.0 || ^7.0",
```

Comment:
But I also think we should drop support for Symfony 3.4 and 4.4 too. Those are officially unmaintained anyway, so it's time to drop it. And I think we can bump 5 & 6 to 5.4 & 6.4 - I hope it will help us to make tests green easier

So, the ideal Symfony support version constraints should be: `^5.4 || ^6.4 || ^7.0` here and in other places.

Wdyt?

Reply:
Yes, I also thought something (see remark in PR message) similar, but I thought I'd await your opinion about this since you have more in depth knowledge of the repo. 

I'll update the PR.

Reply:
Ok, I managed to fix some of the tests, however some of the other ones I'm not really sure what to do with. For the the remaining errors and failures it's related to my remark in the PR message. I'll provide an example:

In some of the code, the used annotations are in the middle of variables
```
        $builder
            ->add('foobar', TextType::class, array(
                /** @Desc("Foobar:") */
                'label' => 'test_label',
            ));
```

But if you try to put an attribute there it will give syntax errors. Furthermore, attributes only seem to be allowed in front of class/function definitions and class properties. 

I'm not aware of any workarounds for this, so I'm not really sure how one would want to support this kind of functionality using attributes. Any of you have an idea?

Reply:
I agree, that felt like an obvious thing to try, but sadly that didn't work.  About using 2 separate readers, I'm guessing that's probably possible.

Also did not think about the fact that adding 7.3 back does not allow typing of class properties, so I'm not sure what you would prefer? Want me to remove the typing or bump it to 8.3?

Reply:
I got lost. I'm fine to drop legacy PHP support and keep only >=8.1. Will the 7.3 support help somehow with this specific case?

Reply:
Uhm no, I just put it back because the PHPStan and CSFixer jobs were failing because of it.  

Now they still are, but now because they aren't accepting the typed class properties.

I found the workflow files, updating them to use 8.1 and bumping composer.json too.

Reply:
> Or I wonder if we can continue using PHP annotations in this specific spot maybe?

I've thought some more about this and I'm not sure that make sense, since it's not really a special case but rather a placement of each of the different Annotations/Attributes. So or I'm missing something or it just means you will need to parse all files twice. Once to see if they have annotations, and another to check for attributes. I think it would make more sense to just stick to annotations if you are keen on retaining this functionality.

I would love to hear your both your thoughts on this. @bocharsky-bw @welcoMattic

Reply:
Hm, I think I saw that Symfony deprecated annotations... but probably that's related only to routes/config. It would be cool to move to the attributes completely, but if we can't do it due to some technical restrictions - probably continue using annotations is the correct option.

However, could we provide both ways: attributes and annotations? You say that files will be parsed twice which may lead to performance downgrade? Probably not a big deal for dev mode? I  think most of people will look for attributes now, and probably not many of them will need that specific `@Desc`, so having attributes would be nice. I wonder about others opinion on it

Reply:
OK, so far I think the best strategy is to continue using annotations, nevertheless, the fact they are feature-complete - they still work great for our case. I don't want to complicate this PR with any custom parcers, etc. The best would be to bring Sf7 support faster. I'm a bit sad to lose the work you've already done with Attributes, but probably not a big deal and it will be in the history of this PR so we can easily cherry-pick that later if needed. So, if you can make tests pass with annotations and Sf 7 support - I would be happy to merge it finally.

Reply:
Completely agree, and it's no big deal, it was not that much to begin with. I'll go work on that, and I'll give an update once I'm done. 

P.S. No worries, it would've indeed been nice if it would have been possible with attributes  but annotations are just a bit more flexible. Who knows, maybe it will change it the future :)

Reply:
Thank you! ❤️

Reply:
I think I have some bad news. I think I found what you were talking about.

I was reverting the changes, and when I came across this file `tests/Functional/Visitor/Php/Symfony/ValidationAnnotationTest.php` I was unable to change it back to AnnotationLoader without PHPstorm complaining.

Some furthing digging dug up:

>**Reading annotations in PHP files is deprecated since Symfony 6.4. Also, the [AnnotationLoader](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Serializer/Mapping/Loader/AnnotationLoader.php) was deprecated in Symfony 6.4.**

soooo.... what now?  :)

Reply:
Ah, heck... that's exactly the problem I faced, now I remember it :/

So, I'm not sure about the best solution here, I'm open to ideas

Reply:
:)

Perhaps a way to test this is to change the test from

```php
 public function testExtractAnnotationError()
    {
        $factory = new LazyLoadingMetadataFactory(new AttributeLoader());
        $extractor = new ValidationAnnotation($factory);
        $collection = $this->getSourceLocations($extractor, Resources\Php\Symfony\ValidatorAnnotationError::class);

        $errors = $collection->getErrors();
        $this->assertCount(1, $errors);
    }
```

to 

```php
 public function testExtractAnnotationError()
    {
        $factory = new LazyLoadingMetadataFactory(new AttributeLoader());
        $extractor = new ValidationAnnotation($factory);
        $collection = $this->getSourceLocations($extractor, Resources\Php\Symfony\ValidatorAnnotationError::class);

        $this->assertCount(0, $collection);
    }
```
Since apparently the AttributeReader does not give in errors when putting in non-existing attributes to extract. It rather just results in no attributes.

I'd like to hear what you think about this change. I'll commit and push it so you can look over the changes without me copy pasting just snippets.

Reply:
Ok so I've had some pressing other issues to take care of but this evening I had some spare time to look into this PR. 

I figured out how to run the GitHub actions locally so I was able to fix most of the warnings and errors that the statistical tests threw. However I've got one left that I don't really understand, and I was hoping it makes more sense to you:

```
[Static analysis/PHPStan     ]   ❗  ::error file=src/Visitor/Php/Symfony/AbstractFormType.php,line=,col=0::Ignored error pattern #^Access to an undefined property PhpParser\\Node\:\:\$args\.$# in path /home/antiftw/repos/forks/php-translation-extractor/src/Visitor/Php/Symfony/AbstractFormType.php was not matched in reported errors.
```

I was able to fix a similar error by adding a null check in the code. I tried the same here, but no luck. However a notable difference is that the other error did indicate a line number where it occurred, and this one does not. Also this one includes a part ("was not matched in reported errors") that the other did not.